docs/adrs/020-model-management-abstractions.md

```diff
@@ -0,0 +1,58 @@
+
+* `training`:
+    * Train in a remote `ray` cluster where the `def train` function is executed as a `ray` job
+    * Train from a local `caikit` library python process using a hosted training API in a running instance of `caikit.runtime` in a remote kubernetes cluster
+* `finding`:
+    * Find models in an alternate saving format such as `transformers`
+    * Find models hosted in a remote storage service such as S3
+* `initializing`:
+    * Load the artifacts using `ray` serving and proxy `run` calls to the instance running in the ray cluster
+    * Delegate the model to a hosted `caikit.runtime` instance in a remote cluster
+
+The previous ADRs for [018-shared-backends](018-shared-backends.md) and [019-loader-stack](019-loader-stack.md) had several key problems:
+
+* They attempted to solve these problems by merging them with the `module_backend` abstraction which has caused a lot of confusion since it was adopted.
+* They didn't differentiate between `finding` and `initialization`
+* They relied on an ordered list of configured objects to discover the right loader instance which was hard to configure in a running server since lists are not well supported with environment variables
+    * This also did not allow explicit specification of a loader for a model where mulitple loaders in the list could work, so you could not load one model one way and another a different way
+
+Prior to `018` and `019`, the `module_backend` abstraction was explicitly tasked with performing "internal distribution" (distributing the work of a given module to some framework, requiring specific knowledge of the algorithm implemented by the module). With `018` and `019`, `module_backend` was extended to also support "external distribution" (distributing the whole operation of a given module as a black box without knowledge of the module's algorithm). This was at the heart of what lead to the confusion, so in this ADR we walk that back and split those responsibilities between `module_backend` (responsible for "internal distribution") and `initializer` (responsible for "external distribution").
+
+## Decision
+
+* The `SharedTrainBackend` an `SharedLoadBackend` abstract classes will be removed
+* The logic in `ModelManager` that handles `SharedLoadBackend`s will be removed
+* A new set of abstract classes will be introduced in `caikit.core.model_management`, one each for `ModelTrainer`, `ModelFinder`, and `ModelLoader`
+* The `ModelManager.load` implementation will be refactored to do the following:
+    * Get a unique `finder` by name or value, falling back to a `"default"` instance and attempt to find the model based on `model_path`
+    * If the model is found, get a unique `initializer` by name or value, falling back to a `"default"` instance and attempt to initialize the found model using it.
```

Comment:
Should we add a line here mentioning that a successful finder operation would result in a config object which would contain a module_id of a caikit module that can load that model? To clarify that model to module resolution is responsibility of the finder.